src/components/ActivityCollector/createLinkClick.js

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import {
+  getAbsoluteUrlFromAnchorElement,
+  isSupportedAnchorElement,
+  isDownloadLink,
+  isExitLink
+} from "./utils";
+
+const determineLinkType = (window, config, linkUrl, clickedObj) => {
+  let linkType = "other";
+  if (isDownloadLink(config.downloadLinkQualifier, linkUrl, clickedObj)) {
+    linkType = "download";
+  } else if (isExitLink(window, linkUrl)) {
+    linkType = "exit";
+  }
+  return linkType;
+};
+
+export default (window, config, clickedObject) => {
+  let linkName;
+  let linkType;
+  let clickedObj = clickedObject;
+  let linkUrl = getAbsoluteUrlFromAnchorElement(window, clickedObj);
+  let isValidLink = false;
+  // Search parent elements for an anchor element
+  // TODO: Replace with generic DOM tool that can fetch configured properties
+  while (clickedObj && clickedObj !== document.body && !linkUrl) {
+    clickedObj = clickedObj.parentElement || clickedObj.parentNode;
```

Comment:
This is from legacy code and I believe this is the reason: https://developer.mozilla.org/en-US/docs/Web/API/Node/parentElement#Browser_compatibility. That said I don't know (and have not reproduced) a scenario where parentElement has failed (I did try this). I just don't want to resurface a previous bug.

Reply:
Can we please remove it until we find and understand a case? I would prefer not to have code in the codebase that's not understood or for which a test could not be written.

Reply:
Sorry, I may have been unclear. I think `clickedObj.parentNode` should stay and we should get rid of `clickedObj.parentElement`. I understand `clickedObj.parentElement` may fail, but if it can fail but `clickedObj.parentNode` can't, I don't understand why we try to use `clickedObj.parentElement` at all.

Reply:
It does look like parentNode will always be available.